grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -728,7 +729,46 @@ class QueryService implements Status {
         Collection result = Collections.emptyList()
 
         if (ids.size() > 0) {
-            result = Issue.findAllByIdInList(ids)
+            result = findIssuesSearchItems(ids)
+        }
+        result
+    }
+
+    /**
+     * Find issues and create DTOs to narrow the amount of data used in the search process
+     * @param issueIds
+     * @return
+     */
+    List<Issue> findIssuesSearchItems(ArrayList<Integer> issueIds) {
+        final String query = "SELECT i.id id, " +
+                "i.project_key projectKey, " +
+                "i.type type, " +
+                "i.status status, " +
+                "i.summary summary, " +
+                "i.reporter reporter, " +
+                "i.update_date updated, " +
+                "i.expiration_date expirationDate " +
+                "FROM issue i " +
+                "WHERE i.id IN (" + issueIds.join(",") + ")"
+
+        List<Issue> result = new ArrayList<Issue>()
+
+        getSqlConnection().rows(query).each{
+            Issue issueSearchItem = new Issue()
+
+            issueSearchItem.setId((Integer)it.get("id"))
+            issueSearchItem.setProjectKey(it.get("projectKey").toString())
+            issueSearchItem.setType(it.get("type").toString())
+            issueSearchItem.setStatus(it.get("status").toString())
+            issueSearchItem.setSummary(it.get("summary").toString())
+            issueSearchItem.setReporter(it.get("reporter").toString())
+            if (it.get("expirationDate") != null) {
+                issueSearchItem.setExpirationDate(it.get("expirationDate"))
+            }
+            if (it.get("updated") != null) {
+                issueSearchItem.setUpdateDate(it.get("updated"))
+            }
+            result.push(issueSearchItem)
```

Comment:
Curious about two things. 
1. Why this is an ordered list? The query is not ordered, but you are forcing the dtos into an array list, so why not just a set?
2. Why use `push`? That's a groovy add-on that just does `add` under the hood - you might have better performance with `add`